src/pipeline.rs

```diff
@@ -44,6 +45,22 @@ impl Runnable for Pipeline {
         }
         Ok("".to_string())
     }
+
+    fn as_any(&self) -> &dyn std::any::Any {
+        self
+    }
+
+    fn as_any_mut(&mut self) -> &mut dyn std::any::Any {
+        self
+    }
+
+    fn to_string(&self) -> String {
+        self.commands
+            .iter()
+            .map(|cmd| cmd.to_string())
+            .collect::<Vec<String>>()
+            .join(" | ")
+    }
```

Comment:
Is it worth implementing the [`Display`](https://doc.rust-lang.org/std/fmt/trait.Display.html) trait instead? Same for the others. Could be done in a separate issue maybe.